core/src/main/java/com/novoda/noplayer/PlayerBuilder.java

```diff
@@ -61,31 +65,39 @@ public PlayerBuilder withWidevineClassicDrm() {
      * @see NoPlayer
      */
     public PlayerBuilder withWidevineModularStreamingDrm(KeyRequestExecutor keyRequestExecutor) {
-        return withDrm(DrmType.WIDEVINE_MODULAR_STREAM, keyRequestExecutor);
+        return withDrm(DrmType.WIDEVINE_MODULAR_STREAM, keyRequestExecutor, null);
     }
 
     /**
      * Sets {@link PlayerBuilder} to build a {@link NoPlayer} which supports Widevine modular download DRM.
      *
-     * @param downloadedModularDrm Implementation of {@link DownloadedModularDrm}.
+     * @param keySetId The KeySetId to restore.
      * @return {@link PlayerBuilder}
      * @see NoPlayer
      */
-    public PlayerBuilder withWidevineModularDownloadDrm(DownloadedModularDrm downloadedModularDrm) {
-        return withDrm(DrmType.WIDEVINE_MODULAR_DOWNLOAD, downloadedModularDrm);
+    public PlayerBuilder withWidevineModularDownloadDrm(final KeySetId keySetId) {
+        KeyRequestExecutor keyRequestExecutor = new KeyRequestExecutor() {
+            @Override
+            public byte[] executeKeyRequest(ModularDrmKeyRequest request) throws DrmRequestException {
+                return keySetId.asBytes();
+            }
+        };
+
+        return withDrm(DrmType.WIDEVINE_MODULAR_DOWNLOAD, keyRequestExecutor, keySetId);
     }
 
     /**
      * Sets {@link PlayerBuilder} to build a {@link NoPlayer} which supports the specified parameters.
      *
-     * @param drmType    {@link DrmType}
-     * @param drmHandler {@link DrmHandler}
+     * @param drmType            {@link DrmType}
+     * @param keyRequestExecutor {@link KeyRequestExecutor}
      * @return {@link PlayerBuilder}
      * @see NoPlayer
      */
-    public PlayerBuilder withDrm(DrmType drmType, DrmHandler drmHandler) {
+    public PlayerBuilder withDrm(DrmType drmType, KeyRequestExecutor keyRequestExecutor, KeySetId keySetId) {
```

Comment:
So, when creating we pass a current `KeySetId` and a `KeyRequestExecutor` to  be executed when expired or about to explire, being `no-player` the one determining when `about to expire` evaluates to `true`.

Is it ok that we do it this way? wouldn't different clients of the library have different requirements on regards to what `about to expire` is?

Reply:
This is actually one of the discussions that I had with @zegnus. I wasn't 100% sure if this was the best approach because we could move the `KeySetId` logic higher.

We could have it so they logic in this class is split for `Downloads` and `Streaming`, almost how it was before. Having the client decide when to use one or the other. It means the client would need to import some helper that is able to determine the expiry time of the `KeySetId` and choose appropriately.

I personally don't mind either approach, this was the one we settled on for now. cc @zegnus

Reply:
Also -> b3ddfbe

Reply:
I think it's client responsibility to handle the renewal of the keys, as different DRM systems could have different requirements.

For now we can leave it like this, we should triple think this

Reply:
While I agree on that potentially the decision should be made by the client, after talking with Ryan we decided to go for this approach. The reason is that this piece of code isolates the client from lots of complexity and potential bugs, while at the same time agreeing that the logic is not a fallback but a default behaviour. If the client passes keys, use it, if they are expired, delegate to a requestKeys method (set up by the client).

We can always go through this again in a call